app/common/services/maps.js

```diff
@@ -103,6 +103,76 @@ function Maps(ConfigEndpoint, L, _, CONST) {
             if (config.default_view.baselayer === 'MapQuest') {
                 config.default_view.baselayer = 'streets';
             }
+            return deferred.promise;
+        },
+        reloadMapConfig: function () {
+            return ConfigEndpoint.get({ id: 'map' }).$promise.then(_.bind(function (config) {
+                // Handle legacy layers
+                if (config.default_view.baselayer === 'MapQuest') {
+                    config.default_view.baselayer = 'streets';
+                }
+                if (config.default_view.baselayer === 'MapQuestAerial') {
+                    config.default_view.baselayer = 'satellite';
+                }
+                this.config = config;
+                return this.config;
+            }, this));
+        }
+    };
+
+    var Map = {
```

Comment:
again. It looks like most of this is out of date. In the current code, I think we could just add this to `createMap` and move the code from posts/modify/location.directive.js thus adding the locate control to *all* our maps